hnn_core/tests/test_cell_response.py

```diff
@@ -146,11 +146,16 @@ def test_cell_response(tmpdir):
         'L2_basket': [[test_rate], [0.0]]}
 
     # Write spike file with no 'types' column
-    # Check for gid_ranges errors
-
     for fname in sorted(glob(str(tmpdir.join('spk_*.txt')))):
         times_gids_only = np.loadtxt(fname, dtype=str)[:, (0, 1)]
         np.savetxt(fname, times_gids_only, delimiter='\t', fmt='%s')
+
+    # Check that spike_types are updated according to gid_ranges
+    cell_response = read_spikes(tmpdir.join('spk_*.txt'),
+                                gid_ranges=gid_ranges)
+    assert cell_response.spike_types == spike_types
```

Comment:
Is this what you were thinking @ntolley, a round-trip test to make sure that spike types can be recovered from legacy spike files given that the user passes in `gid_ranges`?

Reply:
In this style yes, but this test still passes without the bug fix? (At least when I just tried it)

I think you'll need to create a situation where `gid_ranges` defines a cell types that never spikes, I'm pretty that's what the if statement is addressing.

Reply:
The issue I encountered when trying to read in a legacy spike file didn't have anything to do with whether or not all the cell types spiked. Also, I just tried this test on master and it failed.

Reply:
Pretty much all this test does is check if `CellResponse.update_types()` has been called in the fringe case where the user defines `gid_ranges`. The logic within `read_spikes()` checks for this, however, sometime in the last few months the function was accidentally set to return the wrong `CellResponse` object.

Reply:
I think this test is fine but it might help to move it after L28 so it's clear that the `spike_types` correspond to the `gid_ranges` provided

Reply:
@jasmainak this is super minor, but we'd also need to move L149-151 up as well, which would separate them from L158-167 and create a similar problem. Should I just make another copy of `gid_ranges` down so that it's easy to read even though redundant?

Reply:
Ah I get it now! Now I just need to figure out what's going on with my local tests...

Reply:
oh I see, let's leave it as it is then.